### DISCUSSION TITLE
Darstellungsfilter in webGISAPI

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#194] [General]
Hallo,
ist es möglich in einer Karte 
![image](https://github.com/user-attachments/assets/b9449dc3-b3d2-4f39-bb33-4ad92c45d27f)
eine Darstellungsfilter zu setzen/löschen(Filterurl und Wert).  Wie zB `service.setServiceVisibility` 
Ich habe in https://docs.webgiscloud.com/dev/API/index.html nix gefunden.
lg

**Comment by gis-eni** [comment]
``service.setServiceVisibility`` setzt die Layer Sichtbarkeit (Layer sichtbar oder nicht sichtbar). Für das setzten von Darstetllungsfilter gibt auch Methoden die offiziell nicht dokumentiert sind, sondern eigentlich nur über die Viewer UI funktionieren.

Hier die inoffizielle Beschreibung:

``` javascript

map.setFilter("service@cmsname~filterid", [{"n":"param1","v":"val1"},{"n":"param2","v":"val2"}]);

``` 

Hier werden über die Karte der Filter gesetzt. Der Filter muss über das oben gezeigt Form übergeben werden ``{service}~{filter}``. Wenn das CMS in der api.config einen Namen hat, muss dieser auch angegeben werden. Dann kommt eine Liste mit Parametern für den Filter:

Beispiel:

``` javascript

{"id":"kataster@my_cms~grundstuecke","args":[{"n":"KG","v":"64154"},{"n":"GNR","v":"10/1"}]);

```

Weiter Funktionen für Filter:

``` javascript

map.unsetFilter("service@cmsname~filterid");
map.hasFilter("service@cmsname~filterid");
map.hasFilters();  // sind irgendwelche Filter gesetzt
map.unsetAllFilters();

```
